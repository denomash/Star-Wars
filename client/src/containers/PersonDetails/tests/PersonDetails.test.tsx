import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import { createHistory, LocationProvider } from "@reach/router";

import PersonDetail from "../PersonDetails";

const renderer = new ShallowRenderer();

let history = createHistory(window);

describe("<PersonDetail />", () => {
  const Component = () => (
    <LocationProvider history={history}>
      <PersonDetail />
    </LocationProvider>
  );

  it("should render and match the snapshot", () => {
    renderer.render(<Component />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
