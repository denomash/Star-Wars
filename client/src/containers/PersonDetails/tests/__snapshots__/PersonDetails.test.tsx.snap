// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PersonDetail /> should render and match the snapshot 1`] = `
<LocationProvider
  history={
    Object {
      "_onTransitionComplete": [Function],
      "listen": [Function],
      "location": Object {
        "hash": "",
        "host": "localhost",
        "hostname": "localhost",
        "href": "http://localhost/",
        "key": "initial",
        "origin": "http://localhost",
        "pathname": "/",
        "port": "",
        "protocol": "http:",
        "search": "",
        "state": null,
      },
      "navigate": [Function],
      "transitioning": false,
    }
  }
>
  <PersonDetails />
</LocationProvider>
`;
